src/s3path.jl

```diff
@@ -1,4 +1,4 @@
-struct S3Path{A<:AbstractAWSConfig} <: AbstractPath
+struct S3Path{A<:Union{Nothing, AbstractAWSConfig}} <: AbstractPath
```

Comment:
Shouldn't be breaking but there may be some incompatibilities with packages that dispatch on `S3Path{<:AbstractAWSConfig}`